internal/provider/data_source_http.go

```diff
@@ -350,6 +356,24 @@ func (d *httpDataSource) Read(ctx context.Context, req datasource.ReadRequest, r
 		}
 	}
 
+	parsedRequestURL, err := url.Parse(requestURL)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error parsing URL",
+			"An unexpected error occurred while parsing the request URL: "+err.Error(),
+		)
+		return
+	}
+
+	if d.provider.Hostname == parsedRequestURL.Hostname() {
+		for name, header := range d.provider.RequestHeaders {
+			request.Header.Set(name, header)
+			if strings.ToLower(name) == "host" {
+				request.Host = header
+			}
+		}
+	}
```

Comment:
Callout: ☝️ Ideally (at least for my use case), these provider-level headers are regarded as sensitive and therefore not logged in plain text if/when `TF_DEBUG=1`, etc. I haven't dug too deep on whether they currently are.